integration-tests/main/src/test/java/io/quarkus/it/main/ParameterResolverTest.java

```diff
@@ -66,4 +73,27 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
         }
     }
 
+    public static class SomeSerializableParameterResolver implements ParameterResolver {
+
+        @Override
+        public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+                throws ParameterResolutionException {
+            return SomeSerializable.class.getName().equals(parameterContext.getParameter().getType().getName());
+        }
+
+        @Override
+        public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+                throws ParameterResolutionException {
+            return new SomeSerializable("foo");
+        }
+    }
+
+    public static class SomeSerializable implements Serializable {
+
+        private final String value;
```

Comment:
Can you add another serializable class as a field?

Reply:
Will do!

Reply:
Added. I also adjusted the visibility of those `DeepClone` implementations since now only the `Provider` shall create them.